chalice/deploy/deployer.py

```diff
@@ -459,15 +459,38 @@ def _create_rest_api_model(self,
                 handler_name=auth.handler_string, stage_name=stage_name,
             )
             authorizers.append(auth_lambda)
+        policy = config.api_gateway_policy
+        if (config.api_gateway_endpoint_type == 'PRIVATE' and not policy):
```

Comment:
I wonder if we should be generating this policy instead of when `config.api_gateway_endpoint_type == 'PRIVATE'` to when `config.api_gateway_endpoint_vpce` is not falsy? I'm mainly asking to see if we want to be able to support setting `api_gateway_endpoint_vpce` without setting an `api_gateway_endpoint_type`? This will allow users to autogenerate a policy for them without forcing the endpoint to be a private.

Reply:
the auto gen'd policy and vpce doesn't really have meaning without configuring the endpoint to being private.

Reply:
I just did some more diving into this... According to the docs: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-resource-policies-aws-condition-keys.html, you cannot specify a vpce-based policy if the APIG API is not private. Given that is the case, let's keep the logic the way it is.